uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/workbench/panels/impl/PlaceManagerTest.java

```diff
@@ -422,6 +422,16 @@ public void testNormalCloseExistingScreenActivity() throws Exception {
         assertFalse(placeManager.getActivePlaceRequests().contains(kansas));
     }
 
+    @Test
+    public void testClosePlaceAlwaysCloseActivityBeforeDestroy() {
+        when(kansasActivity.isType(any())).thenReturn(false);
+
+        placeManager.closePlace(kansas);
+
+        verify(kansasActivity).onClose();
+        verify(activityManager).destroyActivity(kansasActivity);
```

Comment:
Would it be possible to verify the order of calls too? Like that `onClose()` was really executed before `destroy` was called? Not a hard req, just thinking that it could give better coverage.

Reply:
good point! I sent a new commit to cover this, thank you!